lib/Doctrine/Deprecations/Deprecation.php

```diff
@@ -72,15 +70,49 @@ class Deprecation
      */
     public static function trigger(string $package, string $link, string $message, ...$args): void
     {
-        // Do not trigger this deprecation if it is temporarily ignored,
-        // because it is expected to be called for 1 or more times.
-        if (array_key_exists($link, self::$temporarilyIgnoredLinks)) {
-            self::$temporarilyIgnoredLinks[$link]--;
+        if (array_key_exists($link, self::$ignoredLinks)) {
+            self::$ignoredLinks[$link]++;
+        } else {
+            self::$ignoredLinks[$link] = 1;
+        }
+
+        if (self::$deduplication === true && self::$ignoredLinks[$link] > 1) {
+            return;
+        }
+
+        // do not move this condition to the top, because we still want to
```

Comment:
I believe, in production, this method should return as early as possible.

Reply:
Yes, i agree. 

I started that in the follow up https://github.com/doctrine/deprecations/pull/24 - though I have an idea to simplify that even more.